lib/chef/azure/commands/enable.rb

```diff
@@ -81,20 +79,41 @@ def enable_chef
     @exit_code
   end
 
-  def install_chef_service
-    @exit_code, error_message = ChefService.new.install(@azure_plugin_log_location)
-    if @exit_code == 0
-      report_status_to_azure "chef-service installed", "success"
-    else
-      report_status_to_azure "chef-service install failed - #{error_message}", "error"
-    end
-    @exit_code
+  def load_chef_service_interval
+    value_from_json_file(handler_settings_file, 'runtimeSettings', '0', 'handlerSettings', 'publicSettings', 'chef_service_interval')
   end
 
   def enable_chef_service
-    @exit_code, error_message = ChefService.new.enable(@chef_extension_root, bootstrap_directory, @azure_plugin_log_location)
+    chef_service = ChefService.new
+    chef_service_interval = load_chef_service_interval
+
+    if chef_service_interval.empty?
+      @exit_code, error_message = chef_service.enable(
+        @chef_extension_root,
+        bootstrap_directory,
+        @azure_plugin_log_location
+      )
+    else
+      if chef_service_interval.to_i == 0
+        @exit_code, error_message = chef_service.disable(
+          @azure_plugin_log_location,
+          bootstrap_directory,
+          chef_service_interval.to_i
+        )
+      else
+        @exit_code, error_message = chef_service.enable(
+          @chef_extension_root,
+          bootstrap_directory,
+          @azure_plugin_log_location,
+          chef_service_interval.to_i
+        )
+      end
+    end
+
     if @exit_code == 0
       report_status_to_azure "chef-service enabled", "success"
+    elsif @exit_code == -1

```

Comment:
We shouldn't use a non 0 exit for a succesful disable. I think you can set a flag above here to determine if you are disabling or enabling successfully.  I'd also suggest doing away with the exit_code alltogether and create a propper exception that you can rescue here but I'll leave that as optional.